pkg/networkservice/common/connect/networkservice_client.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"context"
+	"net/url"
+
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/heal"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+)
+
+// NewNetworkServiceClient - adapts connect server chain to client
+func NewNetworkServiceClient(ctx context.Context, connectTo *url.URL, clientOpts ...Option) networkservice.NetworkServiceClient {
+	var opts = &connectServer{}
+	for _, opt := range clientOpts {
+		opt(opts)
+	}
+	return adapters.NewServerToClient(
+		chain.NewNetworkServiceServer(
+			clienturl.NewServer(connectTo),
+			heal.NewServer(ctx, nil),
+			NewServer(
+				ctx,
+				func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient {
+					return chain.NewNetworkServiceClient(
+						append(
+							opts.clientAdditionalFunctionality,
+							heal.NewClient(ctx, networkservice.NewMonitorConnectionClient(cc)),
```

Comment:
Please remove `heal` from here, it should come as a part of additional functionality.

Reply:
But we need `cc` for heal.NewClient(...). Probably we need to use extra option `WithHeal`?

Reply:
OK, I see. So probably it would be better to use here `ClientFactory` instead of `additionalFunctionality` if so. `connect` shouldn't now anything about `heal`.